adaptive/learner/balancing_learner.py

```diff
@@ -119,27 +119,34 @@ def strategy(self, strategy):
                 ' strategy="npoints", or strategy="cycle" is implemented.'
             )
 
+    def _ask_all_learners(self, total_points):
+        to_select = []
+        for index, learner in enumerate(self.learners):
+            # Take the points from the cache
+            if index not in self._ask_cache:
+                self._ask_cache[index] = learner.ask(n=1, tell_pending=False)
+            points, loss_improvements = self._ask_cache[index]
+            if not points:  # cannot ask for more points
+                return to_select
```

Comment:
There are a couple of things I don't understand here:

+ seems this `return` should be a `continue`; just because learner `i` could not give any more points does not mean that no other learners can give any!
+ I cannot see when this branch will ever be executed. `learner.ask(1)` is guaranteed to return a point. At the moment there is no way for a learner to indicate that it has "no more points". If a learner returns no points then it is in violation of the API and other stuff is liable to break